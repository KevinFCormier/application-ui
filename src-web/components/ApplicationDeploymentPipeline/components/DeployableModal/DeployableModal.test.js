/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

const React = require('react')
const renderer = require('react-test-renderer')
const DeployableModal = require('../DeployableModal').default

describe('DeployableModal', () => {
  it('DeployableModal renders correctly.', () => {
    const tree = renderer.create(<DeployableModal />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
