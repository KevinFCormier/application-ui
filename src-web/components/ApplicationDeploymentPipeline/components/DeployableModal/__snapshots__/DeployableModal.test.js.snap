// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployableModal DeployableModal renders correctly. 1`] = `
<div
  id="DeployableModal"
>
  <div
    className="bx--modal bx--modal-tall modalAddRepo"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onTransitionEnd={undefined}
    role="presentation"
    tabIndex={-1}
  >
    <div
      aria-label={undefined}
      className="bx--modal-container"
      role="dialog"
    >
      <div
        className="bx--modal-header"
      >
        
        <h2
          className="bx--modal-header__heading"
        >
          
        </h2>
        <button
          className="bx--modal-close"
          onClick={[Function]}
          type="button"
        >
          <svg
            alt="close the modal"
            aria-label="close the modal"
            className="bx--modal-close__icon"
            fill={undefined}
            fillRule="evenodd"
            height="10"
            name={undefined}
            role="img"
            style={undefined}
            viewBox="0 0 10 10"
            width="10"
          >
            <title>
              close the modal
            </title>
            <path
              d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
            />
          </svg>
        </button>
      </div>
      <div
        className="bx--modal-content"
      >
        <div
          className="channelGridContainer"
        >
          <div
            className="subscriptionInfoClass"
          >
            <div
              className="subHeader"
            >
              <div
                className="bold"
              >
                Subscription Name
              </div>
              subName
            </div>
            <div
              className="innerContent"
            >
              <div
                className="placement"
              >
                <div>
                  <div
                    className="label"
                  >
                    Placement
                  </div>
                </div>
                <div>
                  <div
                    className="label-indented"
                  >
                    Clusters
                  </div>
                  <div
                    className="value"
                  >
                    cluster1
                    cluster2
                    cluster3
                  </div>
                </div>
                <div>
                  <div
                    className="label-indented"
                  >
                    Label
                  </div>
                  <div
                    className="value"
                  >
                    label1
                    label2
                    label3
                  </div>
                </div>
              </div>
              <div
                className="versions"
              >
                <div>
                  <div
                    className="label"
                  >
                    Deployable versions
                  </div>
                  <div
                    className="value"
                  >
                    version2
                  </div>
                </div>
              </div>
              <div
                className="update"
              >
                <div>
                  <div
                    className="label"
                  >
                    Rolling update
                  </div>
                  <div
                    className="value"
                  >
                    50%
                  </div>
                </div>
              </div>
              <button
                className="editSubscriptionButton"
                onClick={[Function]}
              />
            </div>
          </div>
          <div
            className="channelInfoClass"
          >
            <div
              className="subHeader"
            >
              <div
                className="bold"
              >
                Channel
              </div>
              chanName
            </div>
            <div
              className="innerContent"
            >
              <div
                className="conditions"
              >
                <div
                  className="label"
                >
                  Conditions
                </div>
                <div
                  className="value"
                >
                  condition1
                </div>
                <div
                  className="value"
                >
                  condition2
                </div>
                <div
                  className="value"
                >
                  condition3
                </div>
              </div>
            </div>
          </div>
          <div
            className="deployableInfoClass"
          >
            <div
              className="subHeader"
            >
              <div
                className="bold"
              >
                Deployable
              </div>
              deployableName
            </div>
            <div
              className="innerContent"
            >
              <div
                className="conditions"
              >
                <div>
                  <div
                    className="label"
                  >
                    Helm chart
                  </div>
                  <div
                    className="value"
                  >
                    --
                  </div>
                </div>
                <div>
                  <div
                    className="label"
                  >
                    Deployable Conditions
                  </div>
                  <div
                    className="value"
                  >
                    passed testing
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="bx--modal-footer"
      >
        <div
          className="bx--modal__buttons-container"
        >
          <button
            className="bx--btn bx--btn--secondary"
            disabled={false}
            onClick={[Function]}
            tabIndex={0}
            type="button"
          />
          <button
            className="bx--btn bx--btn--primary"
            disabled={false}
            inputref={
              Object {
                "current": null,
              }
            }
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
